airbyte-webapp/src/components/connection/ConnectionForm/formConfig.tsx

```diff
@@ -144,102 +144,96 @@ const createConnectionValidationSchema = (
         then: yup.string().trim().required("form.empty.error"),
       }),
       prefix: yup.string(),
-      syncCatalog: isNewTableDesignEnabled
-        ? yup.object({
-            streams: yup.array().of(
-              yup.object({
-                id: yup
-                  .string()
-                  // This is required to get rid of id fields we are using to detect stream for edition
-                  .when("$isRequest", (isRequest: boolean, schema: yup.StringSchema) =>
-                    isRequest ? schema.strip(true) : schema
-                  ),
-                stream: yup.object(),
-                config: yup.object({
-                  selected: yup.boolean(),
-                  syncMode: yup.string(),
-                  destinationSyncMode: yup.string(),
-                  primaryKey: yup
-                    .array()
-                    .of(yup.array().of(yup.string()))
-                    .when(["syncMode", "destinationSyncMode", "selected"], {
-                      is: (syncMode: SyncMode, destinationSyncMode: DestinationSyncMode, selected: boolean) =>
-                        syncMode === SyncMode.incremental &&
-                        destinationSyncMode === DestinationSyncMode.append_dedup &&
-                        selected,
-                      then: yup.array().of(yup.array().of(yup.string())).min(1, "form.empty.error"),
-                    }),
-                  cursorField: yup
-                    .array()
-                    .of(yup.string())
-                    .when(["syncMode", "destinationSyncMode", "selected"], {
-                      is: (syncMode: SyncMode, destinationSyncMode: DestinationSyncMode, selected: boolean) =>
-                        (destinationSyncMode === DestinationSyncMode.append ||
-                          destinationSyncMode === DestinationSyncMode.append_dedup) &&
-                        syncMode === SyncMode.incremental &&
-                        selected,
-                      then: yup.array().of(yup.string()).min(1, "form.empty.error"),
-                    }),
-                }),
-              })
-            ),
-          })
-        : yup.object({
-            streams: yup.array().of(
-              yup.object({
-                id: yup
-                  .string()
-                  // This is required to get rid of id fields we are using to detect stream for edition
-                  .when("$isRequest", (isRequest: boolean, schema: yup.StringSchema) =>
-                    isRequest ? schema.strip(true) : schema
-                  ),
-                stream: yup.object(),
-                config: yup
-                  .object({
+      syncCatalog: yup.object({
```

Comment:
There was a lot of code duplication here, so the `isNewTableDesignEnabled` check is moved down to the config section, since this is where the two differ (the new table validates the pk and cursor individually)